docs/dev/getting_started.rst

```diff
@@ -62,46 +63,68 @@ Checking out the code
+
+You can learn more about using `virtualenv <https://virtualenv.pypa.io/en/stable/userguide/>`_, or follow these basic instructions:
+
+Initial setup, performed once:
 
 .. code-block:: bash
 
-  $ sudo pip install virtualenv
-  $ cd my_kolibri_project_folder
-  $ mkvirtualenv kolibri
-  $ workon kolibri 
-  (kolibri)$ 
+  $ sudo pip install virtualenv  # install virtualenv globally
+  $ mkdir ~/.venvs               # create a common directory for multiple virtual environments
+  $ virtualenv ~/.venvs/kolibri  # create a new virtualenv for Kolibri dependencies
+
 
 .. note::
-  In this document we use the name ``kolibri``, but you can name the virtual environment however you wish.
+
+  We create the virtualenv `outside` of the Kolibri project folder. You can choose another location than ``~/.venvs/kolibri`` if desired.
+
+To activate the virtualenv in a standard Bash shell:
+
+.. code-block:: bash
+
+  $ source ~/.venvs/kolibri/bin/activate  # activate the venv
+
+Now, any commands run with ``pip`` will target your virtualenv rather than the global Python installation.
+
+To deactivate the virualenv, run the command below. Note, you'll want to leave it activated for the remainder of project setup!
+
+.. code-block:: bash
+
+  $ deactivate
+
 
 .. tip::
-  Linux users might also consider using `virtualenvwrapper <http://virtualenvwrapper.readthedocs.io/en/latest/index.html>`_.
+
+  * Users of Windows and other shells such as Fish should read the `guide <https://virtualenv.pypa.io/en/stable/userguide/>`_ for instructions on activating.
+  * If you set the ``PIP_REQUIRE_VIRTUALENV`` environment variable to ``true``, pip will only install packages when a virtualenv is active. This can help prevent mistakes.
+  * Bash users might also consider using `virtualenvwrapper <http://virtualenvwrapper.readthedocs.io/en/latest/index.html>`_, which simplifies the process somewhat.
+
 
 
 Install Project Dependencies
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 .. note::
 
-  If you're *not* using a Python virtual environment, you may need to use ``sudo`` with the ``pip install`` commands below. ``yarn install`` automatically isolates project dependencies and works without ``sudo``.
-
-#. Install ``yarn`` according the `instructions specific for your OS <https://yarnpkg.com/en/docs/install/>`_.
+  Make sure your virtualenv is active!
 
-#. To install Kolibri project-specific dependencies make sure you're in the ``kolibri`` directory and run:
+To install Kolibri project-specific dependencies make sure you're in the ``kolibri`` directory and run:
 
   .. code-block:: bash
 
     # Python requirements
     (kolibri)$ pip install -r requirements.txt
     (kolibri)$ pip install -r requirements/dev.txt
 
-    # Node.js dependencies
-    (kolibri)$ yarn install
-
     # Kolibri Python package in 'editable' mode, so your installation points to your git checkout:
     (kolibri)$ pip install -e .
 
+    # Javascript dependencies
+    (kolibri)$ yarn install
+
+
```

Comment:
order doesn't matter, but I thought it was cleaner to group all the `pip` actions into one block